forch/authenticator.py

```diff
@@ -23,22 +23,34 @@
 
 class Authenticator:
     """Authenticate devices using MAB/dot1x"""
-    def __init__(self, radius_ip, radius_port, radius_secret, auth_callback=None):
+    def __init__(self, auth_config, auth_callback=None):
         self.auth_map = self._get_auth_map()
         self.radius_query = None
         self.sessions = {}
         self.auth_callback = auth_callback
-        if radius_ip and radius_port and radius_secret:
-            Socket = collections.namedtuple(
-                'Socket', 'listen_ip, listen_port, server_ip, server_port')
-            socket_info = Socket('0.0.0.0', 0, radius_ip, radius_port)
-            self.radius_query = r_query.RadiusQuery(
-                socket_info, radius_secret, self.process_radius_result)
-            threading.Thread(target=self.radius_query.receive_radius_messages, daemon=True).start()
+        radius_info = auth_config.get('radius_info')
+        if not radius_info:
```

Comment:
what does it mean for authentication to exist when this is not defined?   Usually better to fail hard, e.g. not allow the creation of an unconfigured authenticator object, unless there's a structural reason for it.

Reply:
Done.